packages/mui-base/src/SelectUnstyled/useSelect.ts

```diff
@@ -50,12 +48,18 @@ function useSelect<TValue>(props: UseSelectParameters<TValue>) {
   const listboxRef = React.useRef<HTMLElement | null>(null);
   const listboxId = useId(listboxIdProp);
 
-  const [value, setValue] = useControlled({
-    controlled: valueProp,
-    default: defaultValue,
-    name: 'SelectUnstyled',
-    state: 'value',
-  });
+  let defaultValue = defaultValueProp;
+  if (valueProp === undefined && defaultValueProp === undefined) {
+    defaultValue = multiple ? [] : null;
+  }
+
+  const optionsMap = React.useMemo(() => {
```

Comment:
I wouldn't go into this rabbit hole unless there's a significant push from the community.
If someone really really needs non-unique values, it can be achieved by boxing them in an object, so the Select compares their references, not values. We can document this as a workaround.
See https://codesandbox.io/s/charming-bose-ib54rm?file=/src/BaseSelect.js